packages/atoms/src/SearchInput.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { TextInput, Pressable, View } from 'react-native';
+import { Box } from './Box';
+import { Icon } from './Icon/Icon';
+
+const MAX_LENGTH = 255;
+
+type InputProps = {
+    value: string;
+    onChange: (value: string) => void;
+    placeholder: string;
+    isClearable?: boolean;
+    isDisabled?: boolean;
+};
+
+type InputStyleProps = {
+    isFocused: boolean;
+};
+
+const inputStyle = prepareNativeStyle(utils => ({
+    flex: 1,
+    fontSize: 16,
+    color: utils.colors.gray700,
+    height: '100%',
+    marginLeft: utils.spacings.md,
+}));
+
+const clearIconStyle = prepareNativeStyle(utils => ({
+    backgroundColor: utils.colors.gray500,
+    height: 20,
+    width: 20,
+    alignItems: 'center',
+    justifyContent: 'center',
+    borderRadius: utils.borders.radii.round,
+}));
+
+const inputWrapperStyle = prepareNativeStyle<InputStyleProps>((utils, { isFocused }) => ({
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    height: 48,
+    borderWidth: 1,
+    borderColor: utils.colors.gray200,
+    backgroundColor: utils.colors.gray200,
+    borderRadius: utils.borders.radii.basic,
+    paddingLeft: 14,
+    paddingRight: 14.25,
+    extend: [
+        {
+            condition: isFocused,
+            style: {
+                borderColor: utils.colors.gray500,
+            },
+        },
+    ],
+}));
+
+export const SearchInput = ({
+    value,
+    onChange,
+    placeholder,
+    isClearable = false,
+    isDisabled = false,
+}: InputProps) => {
+    const { applyStyle } = useNativeStyles();
+    const [isFocused, setIsFocused] = useState<boolean>(false);
+
+    const handleClear = () => {
+        // Wanted behaviour for clearing text from input.
+        onChange('');
+    };
+
+    return (
+        <Box style={applyStyle(inputWrapperStyle, { isFocused })}>
+            <View>
+                <Icon type="search" color="gray600" />
+            </View>
+            <TextInput
+                value={value}
+                onChangeText={onChange}
+                placeholder={placeholder}
+                maxLength={MAX_LENGTH}
```

Comment:
avoid unnecessary guards etc. it should be added only when needed

Reply:
[8dde2c5](https://github.com/trezor/trezor-suite/pull/5433/commits/8dde2c5f1c1693cc984724aa9c04277bf7047375)